packages/http-server-csharp/src/lib/service.ts

```diff
@@ -268,30 +272,120 @@ export async function $onEmit(context: EmitContext<CSharpServiceEmitterOptions>)
       const decl = this.emitter.result.declaration(
         className,
         code`${this.#generatedFileHeader}
 
       ${this.#emitUsings()}
-      
       namespace ${namespace} {
 
-      ${doc ? `${formatComment(doc)}\n` : ""}${`${attributes.map((attribute) => attribute.getApplicationString(this.emitter.getContext().scope)).join("\n")}${attributes?.length > 0 ? "\n" : ""}`}public partial class ${className} ${
-        model.baseModel ? code`: ${this.emitter.emitTypeReference(model.baseModel)}` : ""
-      } {
-      ${this.emitter.emitModelProperties(model)}
+      ${doc ? `${formatComment(doc)}\n` : ""}${`${attributes.map((attribute) => attribute.getApplicationString(this.emitter.getContext().scope)).join("\n")}${attributes?.length > 0 ? "\n" : ""}`}public partial class ${className} ${baseClass} {
+      ${exceptionConstructor ? `${exceptionConstructor}\n` : ""}${this.emitter.emitModelProperties(model)}
     }
    } `,
       );
 
       return decl;
     }
 
+    getModelExceptionConstructor(program: Program, model: Model, className: string): string {
+      function getDefinedStatusCode() {
+        const statusCodeProperty = new ModelInfo().filterAllProperties(program, model, (p) =>
+          isStatusCode(program, p),
+        );
+
+        if (!statusCodeProperty) return undefined;
+
+        const { type } = statusCodeProperty;
+        switch (type.kind) {
+          case "Union": {
+            const firstVariant = type.variants.values().next().value;
+            return firstVariant?.type.kind === "Number" ? firstVariant.type.value : undefined;
+          }
+          case "Number":
+            return type.value;
+          default:
+            return getMinValue(program, statusCodeProperty) ?? undefined;
+        }
+      }
+
+      if (!isErrorModel(program, model)) {
+        return "";
+      }
+
+      const definedStatusCode = getDefinedStatusCode();
+      const constructor = this.getExceptionConstructorData(program, model);
+
+      return `public ${className}(${constructor.properties}) : base(${definedStatusCode ? definedStatusCode : "default"}${constructor.header ? `, \n\t\t headers: new(){${constructor.header}}` : ""}${constructor.value ? `, \n\t\t value: new{${constructor.value}}` : ""}) 
+        { ${constructor.body ? `\n${constructor.body}` : ""}\n\t}`;
+    }
+
+    getExceptionConstructorData(program: Program, model: Model) {
+      const allProperties = new ModelInfo().getAllProperties(program, model) ?? [];
+      const propertiesWithDefaults = allProperties.map((prop) => {
+        const { defaultValue: typeDefault } = this.#findPropertyType(prop);
+        const defaultValue = prop.defaultValue
+          ? code`${JSON.stringify(serializeValueAsJson(program, prop.defaultValue, prop))}`
+          : typeDefault;
+        return { prop, defaultValue };
+      });
+
+      const sortedProperties = propertiesWithDefaults
```

Comment:
Just a note, nothing to do here.  This is an instance where this is the *same* as resolving properties in other contexts, but we have a bespoke implementation.  We need to think about how to have this resolution done in one place for all of these contexts.

Reply:
BTW, this is a pretty slick bespoke implementation, though